### PR TITLE
Updates test mode instructions copy for card

### PR DESCRIPTION
**Pull request by FangedParakeet**
#### Changes proposed in this Pull Request
This PR updates the test mode instructions for credit cards at checkout following designs here -> pfHfG4-gT-p2#comment-211

The new copy is shorter than the previous copy and there is a new copy icon that should copy the test code number to your clipboard. 

Here's a few examples of what this should look like...

##### Blocks
**Storefront**
Storefront CC

**Deli**
Deli CC

##### Shortcode
**Storefront**
Storefront CC

**Deli**
Deli CC
_Here with a few saved payment methods_

#### Testing instructions

* Please pull the changes in this PR, of course. Visit the checkout and select the credit card payment gateway.
* Please test this PR in both shortcode and blocks checkouts and ensure presentation and functionality is consistent.
* Confirm that copy icon copies test code number to clipboard.
* Confirm that Testing Guide links out to the correct link.
* If possiblem, please test this on multiple themes and confirm that there are no significant issues with display.

#### Dev notes

pfHfG4-gT-p2#comment-211

-------------------

- [x] Run `npm run changelog` to add a changelog file, choose `patch` to leave it empty if the change is not significant. You can add multiple changelog files in one PR by running this command a few times. 
- [x] Covered with tests (or have a good reason not to test in description ☝️)
- [x] Tested on mobile (or does not apply)

**Post merge**

- [ ] Link to testing instructions from [release testing doc](https://github.com/Automattic/woocommerce-payments/wiki/Release-testing-instructions) following [these instructions](https://github.com/Automattic/woocommerce-payments/wiki/How-to-write-good-manual-testing-scenarios) : _Add link here / 'QA Testing Not Applicable'_
- [ ] Add or update [critical flows](https://github.com/Automattic/woocommerce-payments/wiki/Critical-flows) and [testing instructions for critical flows](https://github.com/Automattic/woocommerce-payments/wiki/Testing-instructions-for-critical-flows), if applicable.
- [ ] Add what's changed (description, screenshot, demo videos etc.) to the release announcement post, if applicable.

### CODE REVIEW

**Comment by botwoo** [comment]
### Test the build 

#### Option 1. Jetpack Beta

- Install and activate [Jetpack Beta](https://jetpack.com/download-jetpack-beta/).
- Use this build by searching for PR number `9272` or branch name `fix/checkout-floating-labels-and-test-mode-copy` in your-test.site/wp-admin/admin.php?page=jetpack-beta&plugin=woocommerce-payments          

#### Option 2. Jurassic Ninja - available for logged-in A12s

:rocket: [Launch a JN site with this branch](https://jurassic.ninja/create/?jetpack-beta&shortlived&nojetpack&woocommerce&woocommerce-payments-dev-tools&branches.woocommerce-payments=fix/checkout-floating-labels-and-test-mode-copy) :rocket:

:information_source: Install this [Tampermonkey script](https://github.com/Automattic/woocommerce-payments/tree/develop/bin/wcpay-live-branches) to get more options.

---

Build info: 

- Latest commit: 18210f54c422469f266d13624ebae0c548f0a4f2
- Build time: 2024-08-16 17:09:18 UTC

_Note: the build is updated when a new commit is pushed to this PR._

**Comment by pierorocca** [comment]
Appreciate the incremental approach here @FangedParakeet. Without the previous "**Test mode:**" copy or the new yellow "Test mode" badge the new copy is looking a bit isolated to me. I've reiterated the technical limitation with Stripe in which we cannot add the copy below the card number field. I've sent @nikkivias the attached preview from my Blocks dev site for comment.

image

Also note that on Blocks checkout, if  credit/debit is the only payment method enabled the radio button is no longer present.
image

**Side note for Nikki:** When there's only one payment method active, it's selected by default. The selected/active border is the only visible border. It gives the illusion that the payment section has a heavier weighted border than the other sections. The approach on WooPay might avoid this visual peculiarity.

image

**Comment by pierorocca** [comment]
@FangedParakeet for the time being Nikki is suggesting to continue prefacing the copy with "**Test mode:**" for the time being as an incremental update.

File: includes/class-wc-payments-checkout.php [hunk 1]
```diff
@@ -330,8 +330,8 @@ public function get_enabled_payment_method_config() {
 				/* translators: link to Stripe testing page */
 				$payment_method->get_testing_instructions(),
 				[
-					'strong' => '<strong>',
-					'a'      => '<a href="https://woocommerce.com/document/woopayments/testing-and-troubleshooting/testing/#test-cards" target="_blank">',
+					'span' => '<span class="copy-icon" onclick="navigator.clipboard.writeText(`4242424242424242`)">',
```
Comment (frosso) [inline includes/class-wc-payments-checkout.php]: Unfortunately, this solution doesn't seem to be very accessible.

I'd recommend using a `button` element:
- It has an outline for keyboard focus
- You can navigate it with keyboard, and it is accessible by screen readers
- It has an active style to provide feedback when clicked

I'd also recommend adding an `aria-label` and a `title` attributes - something like "Copy the test card number to your clipboard" and "Copy to your clipboard", respectively (as an example)..

File: includes/payment-methods/class-cc-payment-method.php [hunk 2]
```diff
@@ -70,6 +70,6 @@ public function get_title( string $account_country = null, $payment_details = fa
 	 * @return string
 	 */
 	public function get_testing_instructions() {
-		return __( '<strong>Test mode:</strong> use the test VISA card 4242424242424242 with any expiry date and CVC. Other payment methods may redirect to a Stripe test page to authorize payment. More test card numbers are listed <a>here</a>.', 'woocommerce-payments' );
+		return __( 'Use test card 4242 4242 4242 4242 <span></span> or refer to our <a>Testing Guide</a>.', 'woocommerce-payments' );
```
Comment (frosso) [inline includes/payment-methods/class-cc-payment-method.php]:
```suggestion
		return __( 'Use test card 4242 4242 4242 4242 <span></span> or refer to our <a>testing guide</a>.', 'woocommerce-payments' );
```

(honest question) does "testing guide" need to be capitalized? I personally find it odd to suddenly see the link capitalized - maybe that's just me :D 
Please do let me know!

**Review by frosso — COMMENTED** [review]
While I understand there still are some minor copy adjustments, I took a look at the implementation and added some suggestions to the PR.

File: includes/payment-methods/class-cc-payment-method.php (hunk 2 above)
Comment (pierorocca) [reply includes/payment-methods/class-cc-payment-method.php]: I felt the same seeing it and I'm the one who proposed it. Switch to lower case.

File: includes/class-wc-payments-checkout.php (hunk 1 above)
Comment (FangedParakeet) [reply includes/class-wc-payments-checkout.php]: Cheers for this insight and instruction. Fixed (hopefully) in 0148c12324ee2bda9563761a02876e862568c2c2.

**Review by pierorocca — APPROVED** [review]
Looks good. Thanks for the quick turn and for adding the clipboard copy feature.

**Pull request merged**